### DISCUSSION TITLE
Keeping dependencies lightweight.

### DISCUSSION THREAD

**Discussion by Wooza** [networkx/networkx#4960] [General]
Hi,

I was wondering why the recent 2.6 release added numpy, pandas, scipy and matplotlib as default dependencies?
Usually, I prefer libraries that keep their required dependencies to a minimum.
So pulling in the entire numpy/pandas/scipy/matplotlib stack seems to really blow up the dependencies of this library.
Would it be possible to make them optional dependencies, so users that need the integration with those libraries can add them?
Similar to how the dependency to pydot is currently handled.

Best regards
Florian

**Comment by dschult** [comment]
See #4967 
We are discussing this actively. We have used the optional dependency approach up to now. We may stick with it afterall, but we wanted to make the default include more.  You can install using `pip install networkx --no-deps` for any specific installation. But we need to find a way for other libraries to prescribe their requirements as networkx with `no-deps`. It seems easier to add deps than remove them from the default.
